src/display.jl

```diff
@@ -1,379 +1,405 @@
 
-        display_params.decorations = decorations
+julia> x
+Interval(0.09999999999999999, 0.30000000000000004)
 
-    end
+julia> setformat(:standard; sigdigits = 3)  # Equivalent to `@format standard 3`
+Display parameters:
+  - format: standard
+  - decorations: false
+  - significant digits: 3
 
-    if sigfigs != nothing
-        if sigfigs < 1
-            throw(ArgumentError("`sigfigs` must be `>= 1`"))
-        end
+julia> x
+[0.0999, 0.301]
+```
+"""
+function setformat(format::Symbol = display_params.format;
+        decorations::Bool = display_params.decorations,
+        sigdigits::Int = display_params.sigdigits)
 
-        display_params.sigfigs = sigfigs
+    format ∉ (:standard, :midpoint, :full) && return throw(ArgumentError("`format` must be `:standard`, `:midpoint` or `:full`."))
+    sigdigits < 1 && return throw(ArgumentError("`sigdigits` must be `≥ 1`."))
 
-    end
+    display_params.format = format
+    display_params.decorations = decorations
+    display_params.sigdigits = sigdigits
 
     return display_params
 end
 
 """
-    @format [style::Symbol] [decorations::Bool] [sigfigs::Integer]
+    @format [format::Symbol] [decorations::Bool] [sigdigits::Integer]
 
-The `@format` macro provides a simple interface to control the output format
-for intervals. These options are passed to the `setformat` function.
-It returns the new `DisplayParameters` object.
+Change the format used by `show` to display intervals. These options are passed
+to the `setformat` function.
 
-The arguments may be in any order and of type:
+Possible options:
+- `format` can be:
+    - `:standard`: `[1, 2]`
+    - `:midpoint`: display `x::Interval` in the form "mid(x) ± radius(x)"
+    - `:full`: display the entire bounds regardless of `sigdigits`.
+- `sigdigits`: number (greater or equal to 1) of significant digits to display.
+- `decorations`: display the decorations or not.
 
-- `Symbol`: the output format (`:full`, `:standard` or `:midpoint`)
-- `Bool`: whether to display decorations
-- `Integer`: the number of significant figures
+See also: `setformat`.
 
-E.g.
+# Example
 ```
-julia> x = 0.1..0.3
-@[0.0999999, 0.300001]
+julia> x = interval(0.1, 0.3)  # Default display options
+[0.0999999, 0.300001]
 
-julia> @format full
+julia> @format full  # Equivalent to `setformat(:full)``
 Display parameters:
-- format: full
-- decorations: false
-- significant figures: 6
+  - format: full
+  - decorations: false
+  - significant digits: 6
 
 julia> x
 Interval(0.09999999999999999, 0.30000000000000004)
 
-julia> @format standard 3
+julia> y = DecoratedInterval(x)
+[0.0999, 0.301]
+
+julia> @format standard true 3  # Equivalent to `setformat(:standard; decorations = true, sigdigits = 3)`
 Display parameters:
-- format: standard
-- decorations: false
-- significant figures: 3
+  - format: standard
+  - decorations: true
+  - significant digits: 3
 
-julia> x
-[0.0999, 0.301]
+julia> y
+[0.0999, 0.301]_com
 ```
 """
 macro format(expr...)
```

Comment:
I reverted this change. You are right that it may be better to have a specific PR targeting docstrings.

About the proposition to remove `@format`: I also fail to see why `@format` is more convenient than just calling `setformat` directly. I can make a PR suggesting to remove it.

Reply:
See PR #563.